### PR TITLE
[Reproduce tool] Allow APP_NAME overrides and other fixes.

### PR DESCRIPTION
**Pull request by mbarbella-chromium**
PTAL

### CODE REVIEW

File: src/local/butler/reproduce.py [hunk 1]
```diff
@@ -257,11 +257,15 @@ def _verify_target_exists(build_directory):
             build_directory=build_directory))
 
 
-def _update_environment_for_testcase(testcase, build_directory):
+def _update_environment_for_testcase(testcase, build_directory, application):
```
Comment (inferno-chromium) [inline src/local/butler/reproduce.py:260]: application_override sounds better here. in args.application, that seems ok.

File: src/local/butler/guard.py [hunk 2]
```diff
@@ -24,8 +24,12 @@ def check_virtualenv():
   root_path = os.path.realpath(
       os.path.join(os.path.dirname(__file__), '..', '..', '..'))
   is_in_virtualenv = bool(os.getenv('VIRTUAL_ENV'))
+  is_reproduce_tool = bool(os.getenv('REPRODUCE_TOOL'))
 
-  if not is_in_virtualenv:
+  # Check that we're in a virtual enviornment unless we're running the
+  # reproduce tool. The tool's method of running does not set the VIRTUAL_ENV
+  # environment variable.
+  if not is_in_virtualenv and not is_reproduce_tool:
```
Comment (inferno-chromium) [inline src/local/butler/guard.py:32]: I dont understand why this is an exception for reproduce tool, butler.py reproduce is run in pipenv, see

$ pipenv run printenv | grep VIRTUAL
VIRTUAL_ENV=/usr/local/google/home/aarya/.local/share/virtualenvs/clusterfuzz-qlNIpp8O

Comment (mbarbella-chromium) [reply src/local/butler/guard.py:32]: Did you run this from within a pipenv shell? I don't see the same result.

File: src/local/butler/reproduce.py [hunk 3]
```diff
@@ -257,11 +257,15 @@ def _verify_target_exists(build_directory):
...
 
 
-def _update_environment_for_testcase(testcase, build_directory):
+def _update_environment_for_testcase(testcase, build_directory, application_override):
   """Update environment variables that depend on the test case."""
   commands.update_environment_for_job(testcase.job_definition)
   environment.set_value('JOB_NAME', testcase.job_type)
 
+  # Override app name if explicitly specified.
+  if application:
```
Comment (inferno-chromium) [inline src/local/butler/reproduce.py:266]: s/application/application_override

**Pull request merged**